tests/library/precompiled-dxil.slang

```diff
@@ -0,0 +1,24 @@
+// precompiled-dxil.slang
+
+// A test that uses slang-modules with embedded precompiled DXIL.
+// The test compiles both library slang (export-library.slang) and entrypoint slang (this file) to two slang-modules, each with -embed-dxil options.
+// The result is two slang-modules that are then linked together in a third slangc invocation.
+// Internally, slang does not use the IR in the modules to link, but rather their embedded DXIL.
+// The test passes if there is no errror thrown.
+// TODO: Check if final linkage used only the precompiled dxil.
+
+//TEST(windows):COMPILE: tests/library/export-library.slang -o tests/library/export-library.slang-module -embed-dxil -profile lib_6_6 -incomplete-library
+//TEST(windows):COMPILE: tests/library/precompiled-dxil.slang -o tests/library/precompiled-dxil.slang-module -embed-dxil -profile lib_6_6 -incomplete-library
+//TEST(windows):COMPILE: tests/library/export-library.slang-module tests/library/precompiled-dxil.slang-module -target dxil -entry computeMain -profile cs_6_6 -o tests/library/linked.dxil
+
+extern int foo(int a);
```

Comment:
I would like to point out that we shouldn't be relying on `extern` and `export` symbols for defining the module boundaries.

When we compile a module to dxil, all public symbols in the module should be automatically treated as `export`.

Reply:
Okay, think I had that working in a previous iteration, but at the time there was an issue with the fact that I did not have the entryPoints properly listed in the CodeGenContext and my ComponentType was at the time just a Module which wasn't working until I created a CompositeType with Module+EntryPoints as children.  I should be able to add that back in.

Reply:
Will fix in next change